### ISSUE TITLE
Problem with "fields_under_root"

### ISSUE THREAD

**Issue by vmcano** [elastic/filebeat#220]
I've tried to use the `fields_under_root` option, but there is a problem with it. I've used it to create tags from Filebeat and when I look at them in Elasticsearch, I find that I have just one tag with all the desired tags concatenated.

I've tried the following:
1. I've created a test filter in Logstash with the following lines:
`filter{
  mutate {
    add_tag => ["TAG1", "TAG2", "TAG3"]
  }
}`
2. In Filebeat, I've used `fields_under_root: false` and `tags: filebeat`=> I have 3 tags (TAG1, TAG2, TAG3) and a filed `tags` under the `fields` field. So, the **expected behaviour**.
3. In Filebeat, I've used `fields_under_root: true` and `tags: filebeat`=> I have 1 tags with the content of all expected tags concatenated: filebeatTAG1TAG2TAG3. Here is where the **problem** is.

**Comment by tsg** [comment]
Can you post your filebeat config for point 3, please?

**Comment by vmcano** [comment]
Here you are:

    ################### Filebeat Configuration #################################
    ############################# Filbeat ######################################
    filebeat:
      prospectors:
        -
          paths:
            - /logs/averageTime.log
          ignore_older: 24h
          fields_under_root: true
          fields:
            tags: filebeat
    
    
      registry_file: /filebeat_reg/.filebeat
    
    ############################# Output ##########################################
    output:
      logstash:
        enabled: true
        hosts:
            - logstash:5043
    
    ############################# Logging ######################################
    logging:
      to_files: true
      to_syslog: false
      files:
        path: /var/log/filebeat
        name: filebeat.log
        keepfiles: 10

**Comment by tsg** [comment]
I see. The `tags` field as added under `fields` is a string, so probably when Logstash does the `add_tag` it concatenates all of them in a single string.

If I get it right, you might be able to use the dedicated tags feature that all Beats have. Try removing the `tags` from under fields and adding the following to your `filebeat.yml`:

<pre>
############################# Shipper #########################################
shipper:
  tags: ["filebeat"]
</pre>

This should output an actual `tags` array, which should play better with your Logstash config.

**Comment by vmcano** [comment]
If I assign tags using the `shipper` part of the configuration, I have to assign the same tag to all prospectors. As far as I know, to assign different tags, I have to use the configuration I posted previously.
I tried to use:

    fields:
      tags: ["filebeat"]

But I get the following error: 

```
2015/11/10 13:37:50.056831 main.go:42: CRIT Config error: Error reading config file: YAML config parsing failed on /etc/filebeat/filebeat.yml: yaml: unmarshal errors:
  line 11: cannot unmarshal !!seq into string. Exiting.

```

**Comment by tsg** [comment]
Yeah, that doesn't work. The "fields" are key-value strings. 

I'd then recommend trying something like this: 

In Filebeat:

<pre>
  fields:
    tag: "filebeat"
</pre>

In Logstash:

<pre>
filter{ 
  mutate { 
    add_tag => ["TAG1", "TAG2", "TAG3", "%{fields.tag}"]
  }
}
</pre>

**Comment by vmcano** [comment]
I checked this feature after the new release and it is still not working.
Of course, there are a lot of different ways to solve this in Logstash but I think it would be desirable to do this in Filebeat to reduce the processing Logstash has to carry out.
I would suggest to change the `question` tag by the `bug` tag.

**Comment by urso** [comment]
This is a problem with logstash not filebeat in paticular. Some fields like 'tags' are special in logstash, making it to use concatenation (as implemented in python). Problem with the concatenation operator in jruby is the type tags are appended to. If tag is a string before applying add_tag filter we will get string concatenation.

Using (as you've tried) should help:

    fields:
        tags: ["filebeat"]

There has been a [PR](#506) fixing the config file parsing error. See related [ticket](elastic/filebeat#69). I'm just not sure if this is included in 1.0.1.

**Comment by ruflin** [comment]
Closing this issue as no update recently. In case the problem persists, please open a topic under https://discuss.elastic.co/c/beats/filebeat
